### PR TITLE
Fix #137 - make a CRD-only CSV

### LINKED ISSUE #137: How CRD installation should work for standalone providers
As far as I understand the OLM flow requires that every CRD is `owned` by one OLM package, and other operators can `require` a CRD and OLM will automatically pull that package and install it to satisfy the requirement. See https://github.com/operator-framework/community-operators/pull/625#issuecomment-523947083

Read more in [owned-crds](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#owned-crds) vs [required-crds](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#required-crds).

So what I think is to publish a CSV from this repo that includes just the CRDs, but without an operator deployment, so that noobaa-operator and rook-ceph and others can require it to have the CRD's installed.

Thoughts?

### PR DESCRIPTION
**Pull request by guymguym**

### CODE REVIEW

**Comment by guymguym** [comment]
Take a look @copejon @jeffvance.
Also pending PR for upstream-community-operators: https://github.com/operator-framework/community-operators/pull/656.

Clearly this CSV is better off being generated, but I wanted to fix the issue first.

File: deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          description: Standard object metadata.
+          type: object
+        spec:
+          description: Specification of the desired behavior of the noobaa system.
```
Comment (jeffvance) [inline deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml]: probably want to keep this description provider agnostic.

File: deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+              description: additionalState gives providers a location to set
+                proprietary config values (tenant, namespace, etc)
+              additionalProperties:
+                type: string
+              type: object
+          required:
+            - storageClassName
+          type: object
+        status:
+          description: Most recently observed status of the noobaa system.
```
Comment (jeffvance) [inline deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml]: or any provisioner...

File: deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          description: Standard object metadata.
+          type: object
+        spec:
+          description: Specification of the desired behavior of the noobaa system.
```
Comment (jeffvance) [inline deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml]: ditto on noobaa

File: deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+              description: AdditionalConfig gives providers a location to set
+                proprietary config values (tenant, namespace, etc)
+              additionalProperties:
+                type: string
+              type: object
+          required:
+            - storageClassName
+          type: object
+        status:
+          description: Most recently observed status of the noobaa system.
```
Comment (jeffvance) [inline deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml]: ditto on noobaa

File: deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  - JSONPath: .spec.reclaimPolicy
+    description: ReclaimPolicy
+    name: Reclaim-Policy
+    type: string
+  - JSONPath: .status.phase
+    description: Phase
+    name: Phase
+    type: string
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
```
Comment (jeffvance) [inline deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml]: I know there's a printcolumns pr related to this field, but on the surface it seems that Age is not the same as a create timestamp w/o doing some math.

File: deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml (hunk 1 above)
Comment (guymguym) [reply deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml]: oops. of course

File: deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml (hunk 2 above)
Comment (guymguym) [reply deploy/crds/objectbucket_v1alpha1_objectbucket_crd.yaml]: done.

File: deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml (hunk 4 above)
Comment (guymguym) [reply deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml]: done

File: deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml (hunk 5 above)
Comment (guymguym) [reply deploy/crds/objectbucket_v1alpha1_objectbucketclaim_crd.yaml]: It works correctly.

**Pull request merged**